server/utils/courseParticipationUtils.test.ts

```diff
@@ -176,28 +182,33 @@ describe('CourseParticipationUtils', () => {
         expect(fieldRow).toBeUndefined()
       })
 
-      it('only shows the location in the Setting row when setting location is undefined', () => {
-        const withoutSettingType = {
-          ...courseParticipationWithName,
-          setting: { location: 'Stockport', type: undefined },
-        }
-
-        const { rows } = CourseParticipationUtils.summaryListOptions(withoutSettingType, referralId)
-        const settingRow = getRow(rows, 'Setting')
-
-        expect(settingRow).toEqual({ key: { text: 'Setting' }, value: { text: 'Stockport' } })
-      })
```

Comment:
I guess this condition will be impossible when the API change to make `type` required again is merged, but until then and updates to the UI do we need to keep it? (setting location provided but setting type not - the util code supports this)

Reply:
You can't set the location but not the `type` in the UI, as you need to select the `type`'s conditional radio button to display the input for `location`. I suppose it's not clear from the implementation code though, we could wrap the `if (courseParticipationWithName.setting.location) ` inside the `if (courseParticipationWithName.setting.type) {` but this will change in future so we'll need to update it anyway.

I'll see what stage the API is at when I've made the other changes to this PR though.

Reply:
True